Adopt node 18 & standardise on fnm for Node version management

lb- opened the pull request:
- Resolves https://github.com/wagtail/wagtail/issues/8499
- Node 18 is close to LTS and will be the LTS once the next release goes out https://nodejs.org/en/about/releases/
- Recommending one Node version manager is simpler `fnm` and it also works better for Windows

lb- commented:
@thibaudcolas  getting this error on the CI - does this require someone to log in to approve the Node update? Or do you think I need to specify an exact version?

https://circleci.com/developer/images/image/cimg/node

```
Build-agent version 1.0.137422-1ab6f72e (2022-09-02T14:10:21+0000)
System information:
 Server Version: 20.10.16
 Storage Driver: overlay2
  Backing Filesystem: xfs
 Cgroup Driver: cgroupfs
 Cgroup Version: 1
 Kernel Version: 5.13.0-1023-aws
 Operating System: Ubuntu 20.04.4 LTS
 OSType: linux
 Architecture: x86_64

Starting container circleci/node:18
Warning: No authentication provided, using CircleCI credentials for pulls from Docker Hub.
  image cache not found on this host, downloading circleci/node:18

Error response from daemon: manifest for circleci/node:18 not found: manifest unknown: manifest unknown
``

lb- commented:
ok - I was able to fix this - the circle CI images naming convention has changed.

https://circleci.com/developer/images/image/cimg/node

Scotchester reviewed (approved):
Working for me locally, working in CircleCI, and working on Squash Labs. Thanks, LB!

lb- commented:
Thanks @Scotchester  I have rebased and will merge this in once CI runs again, I also changed to 18.12 not 18.10 (as that's officially when the LTS started).
